src/features/search/bestseller/bestseller.ts

```diff
@@ -15,7 +18,7 @@ export async function bestseller(params: BestsellerOptions): Promise<SearchRespo
     start: 0,
   };
 
-  const queryParams = Object.assign(config, defaults, params);
+  const queryParams = Object.assign(config, defaults, options);
```

Comment:
It will be good to have a function that extracts out whitelisted parameters from the config that's relevant for the API request, if we add configuration keys to the Configuration type, that configures the behavior of the SDK and are not relevant for the API, like a debug flag for example. This code will change to something like this:

`const queryParams = buildQueryParams(config, Object.assign(defaults, options); `

This can be done in the future too, when we add SDK specific configuration, so feel free to ignore it in this PR

Reply:
Yea I think especially the 'whitelisted' part is important here. We are already doing that by indicating through the types what should be in the object. But in javascript you could of course pass in anything. Thats not what we want and its not very secure either. I'm looking to add 'zod' to the project to do schema validation etc to solve this.